LabApiUtilities/src/main/com/microsoft/identity/labapi/utilities/authentication/KeyVaultCertificateBasedAuthenticationClient.java

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.labapi.utilities.authentication;
+
+import com.microsoft.identity.labapi.utilities.authentication.client.IConfidentialAuthClient;
+import com.microsoft.identity.labapi.utilities.authentication.common.CertificateCredential;
+import com.microsoft.identity.labapi.utilities.authentication.common.ClientCertificateMetadata;
+import com.microsoft.identity.labapi.utilities.authentication.common.KeyStoreConfiguration;
+import com.microsoft.identity.labapi.utilities.authentication.msal4j.Msal4jAuthClient;
+import com.microsoft.identity.labapi.utilities.exception.LabApiException;
+import com.microsoft.identity.labapi.utilities.exception.LabError;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import lombok.NonNull;
+
+/**
+ * A an authentication client that can acquire access tokens for Labs KeyVaults (rather than Labs API).
+ */
+public class KeyVaultCertificateBasedAuthenticationClient implements IAccessTokenSupplier {
```

Comment:
You can access secrets through the Labs API. This class fetches secrets from the KeyVault directly. Why is this needed? Because our test config is stored in the `buildautomation` vault, which isn't supported through the API.

Reply:
So you're using the cert to fetch a token from lab api, which you can use to access your keyvault?

Reply:
Correct. As mentioned in your PR, I believe there are 3 types of Labs-based auths:
1. `KeyVaultAuthenticationClient`: fetch tokens for the KeyVault Labs API (i.e. `msidlab.com/api/LabSecret?secret={value}`
2. `LabApiAuthenticationClient`: fetch tokens for other Labs API functionality (e.g. `https://msidlab.com/api/app/{value}`
3. `KeyVaultCertificateBasedAuthenticationClient`: fetch tokens for the KeyVault API directly (not through Labs API) (e.g. `buildautomation.vault.azure.net/secrets/{value}`